front/src/applications/operationalStudies/components/SimulationResults/SimulationResultsMap.tsx

```diff
@@ -176,12 +188,12 @@ const Map: FC<MapProps> = ({ setExtViewport }) => {
       // Found trains including timePosition, and organize them with geojson collection of points
       setOtherTrainsHoverPosition(
         createOtherPoints().map((train) => {
-          const headDistanceAlong = train.head_positions.position / 1000;
-          const tailDistanceAlong = train.tail_positions.position / 1000;
+          const headDistanceAlong = train.head_positions?.position ?? 0 / 1000;
+          const tailDistanceAlong = train.tail_positions?.position ?? 0 / 1000;
```

Comment:
I'm not sure that this will work. You need to add parenthesis. Can you test that this works ?
```suggestion
          const headDistanceAlong = (train.head_positions?.position ?? 0) / 1000;
          const tailDistanceAlong = (train.tail_positions?.position ?? 0) / 1000;
```

Reply:
To me syntax is correct but I've add parenthesis for clear reading